### ISSUE TITLE
Paste in iTerm2 and Terminal on OSX behaves strangely

### ISSUE THREAD

**Issue by unphased** [neovim/neovim#2701]
I find that I have to be NOT in insert mode when I paste (Cmd+V, I am positive that this is in fact *supposed to be* doing nothing more than simply sending the keys I am pasting into the terminal). 

I have to be in normal mode when I invoke paste (somehow successfully pasting the entire pastebuffer, and it leaves me in insert mode). 

If i do it while in insert mode in neovim, something gets removed, seems like it's removing the previous word and then pasting all but the last word.  

I'm really really confused right now. 

Doesn't happen in vanilla vim. Doesnt happen if I SSH from PuTTY (which I know does not support bracketed paste). Does neovim silently enable bracketed paste mode handling? It seems to be the only possible explanation as it would mean that neovim switches it on (changing iTerm's behavior), whereas if I, say, run the libtermkey `demo` testing app, it doesn't switch it on, and i still see what I expect to see. 

I sort of just want to get back the ability to just paste while in insert mode... Actually I thought hard about what I was seeing, and it's consistent with some additional sequence being sent immediately before and after the paste content (which could potentially explain the removal of one word before paste and the missing last pasted word). 

The main problem (considering that I can just continue to use the behavior) is that I have to do pasting in normal mode in one terminal emulator and do it in insert mode (after enabling set paste) in another. For the same machine! Just Makes No Sense.

**Comment by unphased** [comment]
Seems like #2007 seems relevant but I am having some trouble understanding the exact mechanism of what is causing problems when I paste while in insert mode.

One more data point is that it does appear to work properly if I run `neovim -u NONE` so I think that does mean that some recent plugin update broke this for me. :disappointed:

**Comment by unphased** [comment]
Can I request a configuration option that disables bracketed paste mode support altogether? Yes, it's heavyhanded, but I want some vanillavim compatibility.

**Comment by unphased** [comment]
Talk about a goddamn wild goose chase. Turns out this was in my vimrc: 

```
set <F37>=^[^?
inoremap <F37> <C-W>
```
Yes, that is the escape sequence Esc+BS (in neovim this is `<M-BS>`). I think there's something special/funky in neovim with F37 (see #2027), and I seem to have just clobbered something, and caused it to emit chars that is ordinarily bound to the "delete word" action.

This bind was the first one I wrote using the fast-bind technique for vanilla vim -- I used it to make alt+backspace delete a word in insert mode. Seems sensible.
